packages/flutter_tools/lib/src/build_system/targets/linux.dart

```diff
@@ -36,12 +36,11 @@ class UnpackLinux extends Target {
   List<Target> get dependencies => <Target>[];
 
   @override
-  Future<void> build(List<File> inputFiles, Environment environment) async {
+  Future<void> build(Environment environment) async {
     final String basePath = artifacts.getArtifactPath(Artifact.linuxDesktopPath);
-    for (File input in inputFiles) {
-      if (fs.path.basename(input.path) == 'linux.dart') {
```

Comment:
Why is excluding this file no longer needed?

Reply:
Because I used input files that would also grab the source file for the rule itself. Since i'm now looking up the path to the artifact and using the directory listing it shouldn't be in that set.